### PR TITLE
Fix known hosts for wrapper script

### PR DESCRIPTION
**Pull request by jsonar-cpapke**
I noticed an issue where if the version of git is less than 2.3.0, the known hosts were not being used in the git call as they were not being added to the wrapper script.

I believe that the change I made shouldn't affect anyone in any other way than making this work as expected.

I don't think any documentation needs to be updated for this change, as the documentation implies that this should work already.

## Contributor Checklist:

* [x] I have updated the unit tests
* [x] I have created a file in the `master/buildbot/newsfragments` directory (and read the `README.txt` in that directory)
* [x] I have updated the appropriate documentation

### CODE REVIEW

File: master/buildbot/test/unit/steps/test_source_git.py [hunk 1]
```diff
@@ -526,7 +541,11 @@ def test_mode_full_clean_ssh_host_key_1_7(self):
         self.expectOutcome(result=SUCCESS)
         self.expectProperty(
             'got_revision', 'f6ad368298bd941e934a41f3babc827b2aa95a1d', self.sourceName)
-        return self.runStep()
+        yield self.runStep()
+
+        expected = b'#!/bin/sh\nssh -i "%s" -o "UserKnownHostsFile=%s" "$@"\n' \
```
Comment (p12tic) [inline master/buildbot/test/unit/steps/test_source_git.py:546]: Can we use `.format()` for string formatting?

File: master/buildbot/test/unit/steps/test_source_git.py [hunk 2]
```diff
@@ -309,7 +316,10 @@ def test_mode_full_clean_ssh_key_1_7(self):
         self.expectOutcome(result=SUCCESS)
         self.expectProperty(
             'got_revision', 'f6ad368298bd941e934a41f3babc827b2aa95a1d', self.sourceName)
-        return self.runStep()
+        yield self.runStep()
+
+        expected = b'#!/bin/sh\nssh -i "%s" "$@"\n' % (unicode2bytes(ssh_key_path),)
```
Comment (p12tic) [inline master/buildbot/test/unit/steps/test_source_git.py:321]: Can we use `.format()` for string formatting?

**Review by p12tic — CHANGES_REQUESTED** [review]
Overall looks great, thank you!

The only nitpick that I had was to use python3 string formatting functions. I'll merge the PR once this gets fixed.

**Comment by p12tic** [comment]
Also, could you squash some of the commits so that they don't pollute history. E.g. commits that fix breakage created in the same PR could be squashed with the commit that introduce the breakage. Also, please use a prefix for the area in the commit messages, e.g. `git:` or `steps:` in this case.

**Comment by jsonar-cpapke** [comment]
Sorry for the delay on updating this pull request.
I have squashed some of the commits into one. I wasn't sure exactly which commits you wanted prefixed, so I only prefixed the one that affected the code, but not the tests. If you want me to prefix the test commits as well, let me know and I will hopefully be able to get that done! (quicker this time)

File: master/buildbot/test/unit/steps/test_source_git.py (hunk 2 above)
Comment (jsonar-cpapke) [reply master/buildbot/test/unit/steps/test_source_git.py]: Changed to using `.format()`!

File: master/buildbot/test/unit/steps/test_source_git.py (hunk 1 above)
Comment (jsonar-cpapke) [reply master/buildbot/test/unit/steps/test_source_git.py]: Changed to using `.format()`!

**Review by p12tic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**